common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -66,6 +67,16 @@
   @VisibleForTesting
   final ConcurrentMap<AppExecId, ExecutorShuffleInfo> executors;
 
+  /**
+   *  Caches index file information so that we can avoid open/close the index files
+   *  for each block fetch.
+   */
+  private final ShuffleIndexCache shuffleIndexCache;
+
+  // Max number of entries to keep in the index cache.
+  private static final String SPARK_SHUFFLE_SERVICE_INDEX_CACHE_ENTRIES = "spark.shuffle.service.index.cache.entries";
+  private static final int DEFAULT_SPARK_SHUFFLE_SERVICE_INDEX_CACHE_ENTRIES = 1024;
```

Comment:
I don't find a common place where spark.shuffle.service.* configs are defined. For example if you check configs like spark.shuffle.service.enabled or spark.shuffle.service.port, they are defined all over the code base.